rust/lance-index/src/scalar/bitmap.rs

```diff
@@ -30,6 +30,11 @@ use super::{btree::TrainingSource, AnyQuery, IndexStore, ScalarIndex};
 
 pub const BITMAP_LOOKUP_NAME: &str = "bitmap_page_lookup.lance";
 
+const MAX_BITMAP_ARRAY_LENGTH: usize = i32::MAX as usize - 1024 * 1024; // leave headroom
+
+// Read in chunks to avoid overflow one binary array
+const MAX_ROWS_PER_CHUNK: usize = 1_000_000;
```

Comment:
How did you come by this value?  Could it still be too large in some cases?  I think something like 32Ki or 64Ki might be safer.

Reply:
I admit I did not put too much thought into this value. Adjusted